bin/utest/pipe.c

```diff
@@ -0,0 +1,102 @@
+#include <assert.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include <signal.h>
+#include <string.h>
+
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include "utest.h"
+#include "util.h"
+
+static sig_atomic_t signal_delivered;
+
+void sigpipe_handler(int signo) {
+  signal_delivered = 1;
+  return;
+}
+
+int test_pipe_parent_signaled(void) {
+  int pipe_fd[2];
+  signal_delivered = 0;
+  signal(SIGPIPE, sigpipe_handler);
+
+  /* creating pipe */
+  assert(pipe2(pipe_fd, 0) == 0);
+
+  /* forking */
+  pid_t child_pid = fork();
+  assert(child_pid >= 0);
+
+  if (child_pid == 0) { /* child */
+    close(pipe_fd[1]);  /* closing write end of pipe */
+    close(pipe_fd[0]);  /* closing read end of pipe */
+    exit(EXIT_SUCCESS);
+  }
+
+  /* parent */
+  close(pipe_fd[0]); /* closing read end of pipe */
+
+  /* Sync with end of child execution */
+  wait_for_child_exit(child_pid, EXIT_SUCCESS);
+
+  /* This is supposed to trigger SIGPIPE and return EPIPE */
+  int write_ret = write(pipe_fd[1], "hello world\n", 12);
```

Comment:
`write` returns `ssize_t` which is `intptr_t` which should have 8 bytes on AArch64.